controllers/storagecluster/cephcluster.go

```diff
@@ -1549,12 +1558,21 @@ func setDefaultMetadataPoolSpec(poolSpec *rookCephv1.PoolSpec, sc *ocsv1.Storage
 			poolSpec.Replicated.ReplicasPerFailureDomain = defaultReplicatedSpec.ReplicasPerFailureDomain
 		}
 	}
+
 }
 
 // setDefaultDataPoolSpec sets the common pool spec for all data pools as necessary
-func setDefaultDataPoolSpec(poolSpec *rookCephv1.PoolSpec, sc *ocsv1.StorageCluster) {
-	poolSpec.EnableCrushUpdates = true
-	if poolSpec.DeviceClass == "" {
+func (r *StorageClusterReconciler) setDefaultDataPoolSpec(poolSpec *rookCephv1.PoolSpec, sc *ocsv1.StorageCluster, specifiedPoolSpecPath []string) {
+	// Check if enableCrushUpdates is explicitly set in the unstructured obj else set to true
+	if val, found := getValueAtPath(r.unstructuredSC, append(specifiedPoolSpecPath, "enableCrushUpdates")); found {
+		poolSpec.EnableCrushUpdates = val.(bool)
+	} else {
+		poolSpec.EnableCrushUpdates = true
+	}
+	// Check if DeviceClass is explicitly set in the unstructured obj else set to the default device class
+	if val, found := getValueAtPath(r.unstructuredSC, append(specifiedPoolSpecPath, "deviceClass")); found {
+		poolSpec.DeviceClass = val.(string)
+	} else if poolSpec.DeviceClass == "" {
```

Comment:
Seems like lines 1566-1575 is duplicated from the method above. Could this be factored out to a helper method?